x/shield/keeper/withdraw.go

```diff
@@ -87,37 +87,52 @@ func (k Keeper) DequeueCompletedWithdrawQueue(ctx sdk.Context) {
 		store.Delete(withdrawTimesliceIterator.Key())
 	}
 
-	// for each completed withdraw, process adjustments.
 	for _, withdraw := range withdraws {
 		provider, found := k.GetProvider(ctx, withdraw.Address)
 		if !found {
 			panic("provider not found but its collaterals are being withdrawn")
 		}
 
-		// update pool community or CertiK first in case the pool is closed
 		pool, found := k.GetPool(ctx, withdraw.PoolID)
 		if !found {
+			// Pools, collaterals and providers have been updated for closed pools.
 			continue
 		}
-		pool.TotalCollateral = pool.TotalCollateral.Sub(withdraw.Amount)
 		collateral, found := k.GetCollateral(ctx, pool, withdraw.Address)
 		if !found {
 			panic("withdraw collateral not found")
 		}
-		// allow collateral amount to be negative, which could happen when it is locked
-		collateral.Amount = collateral.Amount.Sub(withdraw.Amount)
+
+		// Update withdrawing.
 		collateral.Withdrawing = collateral.Withdrawing.Sub(withdraw.Amount)
-		if collateral.Amount.IsNegative() && len(collateral.LockedCollaterals) == 0 {
+		provider.Withdrawing = provider.Withdrawing.Sub(withdraw.Amount)
+
+		// Update collateral.
+		// It is possible that the withdraw amount exceeds the collateral amount because of locking collaterals by claim proposals.
+		// Do not allow collateral amount to be negative. Set overdraft for this situation.
+		var validWithdrawAmount sdk.Int
+		if collateral.Amount.GTE(withdraw.Amount) {
+			validWithdrawAmount = withdraw.Amount
+		} else {
+			validWithdrawAmount = collateral.Amount
+			collateral.Overdraft = collateral.Overdraft.Add(withdraw.Amount.Sub(collateral.Amount))
```

Comment:
What about calling it `LockedWithdrawal`?

Reply:
Renamed.